OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldaccess/reflection/ReflectionRelatedFieldAccessesAnalysis.scala

```diff
@@ -971,16 +971,21 @@ class MethodHandleInvokeAnalysis private[analyses] (
         if (definition.isMethodHandleConst) {
             definition.asMethodHandleConst.value match {
                 case handle: InstanceFieldAccessMethodHandle =>
-                    matchers += MatcherUtil.retrieveSuitableNonEssentialMatcher[V](
-                        actualParams.flatMap(_.head),
-                        v => new ActualReceiverBasedFieldMatcher(v.value.asReferenceValue)
-                    )
-                    matchers ++= MethodHandlesUtil.retrieveMatchersForMethodHandleConst(
-                        handle.declaringClassType,
-                        handle.name,
-                        handle.fieldType,
-                        isStatic = false
-                    )
+                    val receiver = actualParams.flatMap(_.head)
+                    if (receiver.exists(r => r.value.isReferenceValue)) {
```

Comment:
This check changes the handling in case the `receiver` is `None`: Previously, the retrieval returned an `AllFieldsMatcher` (which adds no filtering). Now, as `exists` will return `false` for the value `None`, all matchers are replaced by a `NoFieldsMatcher`.

In case we have no information about the receiver I am unsure if we should filter out all fields immediately. Only when we know that the value is not a reference value we may immediately remove all fields.